pkg/codegen/pcl/binder_schema_test.go

```diff
@@ -19,3 +22,34 @@ func BenchmarkLoadPackage(b *testing.B) {
 		contract.AssertNoError(err)
 	}
 }
+
+func TestGenEnum(t *testing.T) {
+	t.Parallel()
+	enum := &model.EnumType{
+		Elements: []cty.Value{
+			cty.StringVal("foo"),
+			cty.StringVal("bar"),
+		},
+		Type:  model.StringType,
+		Token: "my:enum",
+		Annotations: []interface{}{
+			enumSchemaType{
+				Type: &schema.EnumType{Elements: []*schema.Enum{{Value: "foo"}, {Value: "bar"}}},
+			},
+		},
+	}
+	safeEnumFunc := func(member *schema.Enum) {}
+	unsafeEnumFunc := func(from model.Expression) {}
+
+	d := GenEnum(enum, &model.LiteralValueExpression{
+		Value: cty.StringVal("foo"),
+	}, safeEnumFunc, unsafeEnumFunc)
+	assert.Nil(t, d)
+
+	d = GenEnum(enum, &model.LiteralValueExpression{
+		Value: cty.StringVal("Bar"),
+	}, safeEnumFunc, unsafeEnumFunc)
+	assert.Equal(t, d.Summary, "\"cty.StringVal(\"Bar\")\" is not a valid value of the enum \"my:enum\"")
+	assert.Equal(t, d.Detail, "Valid members are: [foo bar]")
```

Comment:
`cty.StringVal` shouldn't show up in our error messages. I think `valid members are "foo", "bar"` is nicer then `Valid members are: [foo bar]`. The error messages are part of `pulumi convert`'s UX, so they should be clear, readable and __not communicate implementation details__ (underlying value type, data structure for enum values).